core/replica.go

```diff
@@ -51,6 +52,33 @@ type Replica struct {
 	handleStream messageStreamHandler
 }
 
+// GetReplica creates a new instance of replica node
+func GetReplica(id uint32, configer api.Configer, stack Stack, handler api.ProtocolHandler, logger *logging.Logger) (*Replica, error) {
```

Comment:
This also seems to me to introduce code duplication, maybe adding `api.ProtocolHandler` into `type replicaStack` is more suitable.

Reply:
Good point. This function is almost identical to `New`. If this proposal passes, it would make sense to simply call `GetReplica` from `New`.

Reply:
Yes, adding the ProtocolHandler to replicaStack is a good design choice, but my idea was to preserve the constructor that allowed us to obtain an instance of Replica that would adopt the MinBFT protocol by default, while also providing a more generic constructor that would allow a third party to supply its own protocol. 

The key idea here is to render replica connectivity as decoupled as possible from the ordering protocol, with the goal to make that connectivity re-usable for other projects that might adopt another ordering protocol different from minbft.